keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md

```diff
@@ -773,15 +786,13 @@ in back-to-back releases.
 - `ServiceAccountNodeAudienceRestriction` feature gate implemented in KAS as a beta feature
   - Audience validation is enabled by default for service account tokens requested by the kubelet
 
-#### Post Alpha
-
-- Make sure the feature is compatible with the [Ensure secret pull images KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2535-ensure-secret-pulled-images).
-
 #### Beta
 
-- The implementation works well with the Ensure secret pull images KEP and supports pod image pull policy set to any value.
+- Make the feature compatible with the [Ensure secret pull images KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2535-ensure-secret-pulled-images).
```

Comment:
Going from 1.34 and on, there is a requirement that features must be code complete to be promoted to beta with feature gate on.

https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/5241-beta-featuregate-promotion-requirements#summary

I am bringing this up because it is not clear if you are making this work required for feature promotion.

This feature is also being promoted in this cycle AFAIK so I don't really understand the cross linking of these two features.

Reply:
cc @deads2k @johnbelamaric.

How should we handle this as part of https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/5241-beta-featuregate-promotion-requirements#summary?

I think the gist of this is that we should be functional complete as much as possible when going to beta on. If #5371 slips or this feature slips, I don't really know how to handle this scenario.

Reply:
It seems somewhat impractical to block that at the KEP phase... any dependent KEP would have to add another cycle, but it might be that the feature must revert to alpha before release if the other feature doesn't land?